LiteCore/tests/QueryTest.cc

```diff
@@ -145,6 +145,366 @@ N_WAY_TEST_CASE_METHOD(QueryTest, "Create/Delete Array Index (Multi-level)", "[Q
     CHECK(!succAll);
 }
 
+namespace {
+
+    void AddDocWithJSON(KeyStore* store, slice docID, ExclusiveTransaction& t, const char* jsonBody) {
+        DataFileTestFixture::writeDoc(
+                *store, docID, DocumentFlags::kNone, t,
+                [=](Encoder& enc) {
+                    impl::JSONConverter jc(enc);
+                    if ( !jc.encodeJSON(slice(jsonBody)) ) {
+                        enc.reset();
+                        error(error::Fleece, jc.errorCode(), jc.errorMessage())._throw();
+                    }
+                },
+                false);
+    }
+
+    inline char nextLetter(char a, unsigned n = 1) { return (char)(a + n); }
+
+    // Turning the state as represented by letters into string. It will be used to generate
+    // the property value of the array property.
+    string concat(std::vector<std::pair<char, unsigned>>& letters, int top, const string& docID) {
+        std::stringstream ss;
+        for ( int i = 0; i <= top; ++i ) {
+            if ( i == 0 && !docID.empty() ) ss << docID << "-";
+            if ( i > 0 ) ss << "-";
+            ss << letters[i].first << letters[i].second;
+        }
+        return ss.str();
+    }
+
+    string multiLevelArrayIter(std::vector<std::pair<char, unsigned>>& letters, int& letter, int arrDim,
+                               const string& docID) {
```

Comment:
Can you provide comment with an example of the result that this function generates?